src/wasm/wasm-validator.cpp

```diff
@@ -209,6 +209,13 @@ struct FunctionValidator : public WalkerPass<PostWalker<FunctionValidator>> {
 
   FunctionValidator(ValidationInfo* info) : info(*info) {}
 
+  void validateInModule(Module& wasm, Expression* curr) {
+    auto* oldModule = getModule();
+    setModule(&wasm);
+    walk(curr);
+    setModule(oldModule);
```

Comment:
Is it possible to call this function with a different module than its old module?

Reply:
Not in the current code. But it seemed risky to me to leave the module modified, as then a future refactoring might run into surprising behavior.

But you made me think that it does seem slightly odd to even consider that possible. I pushed a small refactoring now to require a module on creation of a FunctionValidator, which will then never change.